ft(social-account):login via google

oliviertech212 opened the pull request:
##  what does this PR do:
sign in with social account
##  Descriptions
- Implement Google OAuth2 authentication for users to log in
- Authenticate users with Google and create a new account
- Redirect users to /welcome endpoint upon successful authentication
- Allow users to log in again using their Google account
- Generate and document Swagger documentation for frontend team
## How this should be tested
- clone the repo
- type git checkout ft-social-account
- type npm install to install dependencies
- type  npm run migrate:up to run migration
-  type npm run dev to start server
- To test login via google by using browser 
copy and paste `GOOGLE_CLIENT_ID=''
GOOGLE_SECRET=''`. into your  .env file
Note: to test this make sure your app is running on port 
open your browser   then type `http://localhost:port/auth/google`
once you done  click on sign in look into your database
- To test login via google by using swagger
  open your browser   then type `http://localhost:port/docs`
click on Authorize
then copy and paste `GOOGLE_CLIENT_ID=''
GOOGLE_SECRET=''`
Note:remember to check the 
 checkbox and then click Authorize
## Sign in with social account card #9

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
solve conflict

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Commit:
ft(social-account):login via google
-register app
-Implement Google OAuth2 authentication for users to log in
-Authenticate users with Google and create a new account
-Redirect users to /welcome endpoint
-Allow users to log in again using their Google account
[# finishes]

Commit:
ft(social-account):login via google
-register app
-Implement Google OAuth2 authentication for users to log in
-Authenticate users with Google and create a new account
-Redirect users to /welcome endpoint
-Allow users to log in again using their Google account
[# finishes]

sostene12 committed:
ft(signin): signin with email and password

-implement signin with email and password
-implement the corresponding errors return
-implement admin seed auto generate

[finished ft-signin]

sostene12 commented:
hey @oliviertech212  , i have tested but. i'm facing the errors for the swagger , i can't see any thing in the database, then on the browser i get the following error , i have used your credentials
`Error: 
    at Query.run (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/dialects/postgres/query.js:76:25)
    at /Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/sequelize.js:650:28
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at PostgresQueryInterface.insert (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/dialects/abstract/query-interface.js:795:21)
    at User.save (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/model.js:4073:35)
    at Function.create (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/model.js:2280:12)
    at Function.findOrCreate (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/node_modules/sequelize/src/model.js:2392:25)
    at Strategy._verify (/Users/andelarwanda/Desktop/ecommerce-app-champions-bn/src/controllers/Oauthcontroller.js:30:34)`

sostene12 reviewed (commented):
the tests are passing, great job

Commit:
ft(social-account):login via google
-register app
-Implement Google OAuth2 authentication for users to log in
-Authenticate users with Google and create a new account
-Redirect users to /welcome endpoint
-Allow users to log in again using their Google account
-[# finishes]

sostene12 committed:
testing: enpoints testing

-implement tests

sostene12 committed:
ft(signin): signin test

- implement tests
- implement cookies once login is happening

[finishes ft-siginin

sostene12 committed:
ft(signin): signin test

- implement tests
- implement cookies once login is happening

[finishes ft-siginin

sostene12 committed:
ft(signin): signin test

- implement tests
- implement cookies once login is happening

[finishes ft-siginin

oliviertech212 committed:
Resolved merge conflict

wakaflorien reviewed (changes requested):
- branch test coverage should be 80 and above

oliviertech212 committed:
ft(logout):Logout from the app
-logout user
-set middleware for protected route
-documment route with swagger
-test the code
[Finishes #6]

oliviertech212 committed:
fix()login with social account
-generate migration to change googleid and facebookid datatype
-implement cookies
